mne_g3d/viz.py

```diff
@@ -233,3 +237,402 @@ def plot_hemisphere_mesh(vertices,
+
+            morph_folder = 'surf/{0}.curv'.format(hemi)
+
+            morph_path = path.join(subjects_dir, subject, morph_folder)
+
+            brain_vertices = hemi_vertices[hemi]
+            brain_faces = hemi_faces[hemi]
+            _, brain_color = read_morph(morph_path)
+
+            if (brain_vertices is not None) and (brain_faces is not None):
+                brain_vertices = offset_hemi(brain_vertices, hemi)
+
+                _, hemi_mesh = plot_hemisphere_mesh(brain_vertices,
+                                                    brain_faces,
+                                                    brain_color,
+                                                    act_colors=act_colors)
+
+                hemi_meshes[hemi] = hemi_mesh
+
+    if time_viewer:
+        control = ipv.animation_control(tuple(hemi_meshes.values()),
+                                        sequence_length=len(stc.times),
+                                        add=False,
+                                        interval=500)
+
+        slider = control.children[1]
+        slider.readout = False
+        slider.value = time_idx
+        if isinstance(time_label, str):
+            label = widgets.Label(time_label % times[time_idx])
+        elif callable(time_label):
+            label = widgets.Label(time_label(times[time_idx]))
+
+        # hadler for changing of selected time moment
+        def slider_handler(change):
+            # change plot
+            nonlocal cmap
+            nonlocal k
+            nonlocal b
+            time_idx_new = int(change.new)
+            stc_data = morph_mat.dot(stc.data[:, time_idx_new])
+
+            for hemi in hemis:
+                if hemi == 'lh':
+                    data = stc_data[:len(hemi_vertices['lh'])]
+                else:
+                    data = stc_data[len(hemi_vertices['lh']):]
+
+                data = k * data + b
+                np.clip(data, 0, 1)
+                act_color_new = cmap(data)
+                hemi_meshes[hemi].color = act_color_new
+
+            # change label value
+            if isinstance(time_label, str):
+                label.value = time_label % times[time_idx_new]
+            elif callable(time_label):
+                label.value = time_label(times[time_idx_new])
+
+        slider.observe(slider_handler, names='value')
+        control = widgets.HBox((*control.children, label))
+
+        # create a colorbar
+        if colorbar:
+            cbar_data = np.linspace(0, 1, cmap.N)
+            cbar_ticks = np.linspace(dt_min, dt_max, cmap.N)
+            color = np.array((cbar_data, cbar_data))
+
+            colors = cmap(cbar_data)
+            # transform to [0, 255] range taking into account transparency
+            bg_color = 0.5 * np.ones(3)
+            colors = np.array([255 * c[:-1] if c[-1] == 1 else
+                               255 * (c[:-1] * c[-1] + bg_color *
+                               (1 - c[-1])) for c in colors])
```

Comment:
Iteration like this should be avoided if possible for performance (and readability) reasons, can you vectorize this using `ndarray` and broadcasting?

Reply:
(In this case it's probably faster and simpler just to do all subtractions/multiplies rather than checking each alpha value, even if some of those alphas are 1)

Reply:
yep, I guess so